File: 115.LeetCode@74_Search_a_2D_Matrix/115.LeetCode@74_Search_a_2D_Matrix.py
class Solution:
    def searchMatrix(self, matrix, target):
        row = 0
        length = len(matrix)
        col = len(matrix[0]) - 1

        while row < length and col >= 0:
            if matrix[row][col] == target:
                return True
            elif target < matrix[row][col]:
                col -= 1
            else:
                row += 1
        return False

File: 115.LeetCode@74_Search_a_2D_Matrix/115.LeetCode@74_Search_a_2D_Matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int row = 0;
        int length = matrix.length;
        int col = matrix[0].length - 1;

        while (row < length && col >= 0) {
            if (matrix[row][col] == target) {
                return true;
            } else if (target < matrix[row][col]) {
                col--;
            } else {
                row++;
            }
        }
        return false;
    }
}